Frameworks/UIKit.Xaml/Layer.xaml.cpp

```diff
@@ -0,0 +1,122 @@
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+// clang-format does not like C++/CX
+// clang-format off
+#include "pch.h"
+#include "Layer.xaml.h"
+
+#include "ObjCXamlControls.h"
+
+using namespace Microsoft::WRL;
+using namespace Platform;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Media;
+
+namespace UIKit {
+namespace Private {
+namespace CoreAnimation {
+
+DependencyProperty^ Layer::s_layerContentProperty = nullptr;
+DependencyProperty^ Layer::s_sublayerCanvasProperty = nullptr;
+bool Layer::s_dependencyPropertiesRegistered = false;
+
+Layer::Layer() {
+    InitializeComponent();
+
+    // No-op if already registered
+    _RegisterDependencyProperties();
+
+    Name = L"Layer";
+
+    // Default to a transparent background brush so we can accept pointer input
+    static auto transparentBrush = ref new SolidColorBrush(Windows::UI::Colors::Transparent);
+    Background = transparentBrush;
+}
+
+// Accessor for our Layer content
+Image^ Layer::LayerContent::get() {
+    if (!_content) {
+        _content = ref new Image();
+        _content->Name = "LayerContent";
+
+        // Layer content is behind any sublayers
+        Children->InsertAt(0, _content);
+    }
+
+    return _content;
+}
+
+// Accessor for our Layer content
+bool Layer::HasLayerContent::get() {
+    return _content != nullptr;
+}
+
+// Accessor for our SublayerCanvas
+Canvas^ Layer::SublayerCanvas::get() {
+    // We *are* the sublayer canvas
+    return this;
+}
+
+DependencyProperty^ Layer::LayerContentProperty::get() {
+    return s_layerContentProperty;
+}
+
+DependencyProperty^ Layer::SublayerCanvasProperty::get() {
+    return s_sublayerCanvasProperty;
+}
+
+void Layer::_RegisterDependencyProperties() {
+    if (!s_dependencyPropertiesRegistered) {
+        s_layerContentProperty = DependencyProperty::RegisterAttached("LayerContent",
+            Image::typeid,
+            FrameworkElement::typeid,
+            nullptr);
+
+        s_sublayerCanvasProperty = DependencyProperty::RegisterAttached("SublayerCanvas",
+            Canvas::typeid,
+            FrameworkElement::typeid,
+            nullptr);
+
+        s_dependencyPropertiesRegistered = true;
+    }
+}
+
+} /* CoreAnimation */
+} /* Private */
+} /* UIKit */
+
+////////////////////////////////////////////////////////////////////////////////////
+// ObjectiveC Interop
+////////////////////////////////////////////////////////////////////////////////////
+
+// Set one or more layer properties for the specified target xaml element
+UIKIT_XAML_EXPORT void XamlSetFrameworkElementLayerProperties(
```

Comment:
These are the only two properties we support; they map to the attached properties registered above in _RegisterDependencyProperties.  The only other property that we may need to add is Border (if we don't port UIScrollView to Xaml before we implement Border on CALayer).

---
In reply to: [86627070](https://github.com/Microsoft/WinObjC/pull/1318#discussion_r86627070) [](ancestors = 86627070)